File: 0011-container-with-most-water/0011-container-with-most-water.scala
object Solution {
    def maxArea(height: Array[Int]): Int = {

        var max_volume = 0

        var lp = 0
        var rp = height.length - 1

        while(lp<rp){

        val current_volume = math.min(height(lp), height(rp)) * (rp-lp)
        max_volume = math.max(current_volume, max_volume)
        if(height(lp)<=height(rp)){
            lp+=1
        }
        else{
            rp-=1
        }

        }
        max_volume
  }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        lp = 0
        rp = len(height)-1

        max_vol = 0
        while(lp<rp):

            max_vol = max(max_vol, (min(height[rp],height[lp]))*(rp-lp))
            if height[lp] <= height[rp]:
                lp+=1
            elif height[lp] > height[rp]:
                rp-=1

            #print(max_vol, height[lp], height[rp])

        return max_vol